php-frontend/src/test/java/org/sonar/php/highlighter/SymbolHighlighterTest.java

```diff
@@ -22,70 +22,123 @@
 import com.google.common.base.Charsets;
 import com.google.common.collect.ImmutableList;
 import com.sonar.sslr.api.typed.ActionParser;
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.fs.internal.DefaultFileSystem;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.DefaultTextPointer;
+import org.sonar.api.batch.fs.internal.DefaultTextRange;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.batch.sensor.symbol.NewSymbolTable;
 import org.sonar.php.parser.PHPParserBuilder;
 import org.sonar.php.tree.symbols.SymbolTableImpl;
 import org.sonar.plugins.php.api.tree.CompilationUnitTree;
 import org.sonar.plugins.php.api.tree.Tree;
 
-import java.util.Collections;
-import java.util.List;
-
-import static org.fest.assertions.Assertions.assertThat;
-
 public class SymbolHighlighterTest {
 
   private static final ActionParser<Tree> PARSER = PHPParserBuilder.createParser(Charsets.UTF_8);
 
+  private File file;
+
+  private DefaultInputFile inputFile;
+
+  private SensorContextTester context;
+
+  private NewSymbolTable newSymbolTable;
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Before
+  public void setUp() throws IOException {
+    DefaultFileSystem fileSystem = new DefaultFileSystem(tempFolder.getRoot());
+    fileSystem.setEncoding(Charsets.UTF_8);
+    file = tempFolder.newFile();
+    inputFile = new DefaultInputFile("moduleKey",  file.getName())
+      .setLanguage("php")
+      .setType(Type.MAIN);
+    fileSystem.add(inputFile);
+
+    context = SensorContextTester.create(tempFolder.getRoot());
+
+    newSymbolTable = context.newSymbolTable().onFile(inputFile);
+  }
+
   @Test
   public void test_empty_input() throws Exception {
-    List<SymbolHighlightingData> data = getData("<?php ");
+    highlight("<?php ");
 
-    assertThat(data).hasSize(0);
+    checkSymbolExistence(1, 1, false);
   }
 
   @Test
   public void test_no_usages() throws Exception {
-    List<SymbolHighlightingData> data = getData("<?php   $a = 1; ");
+    highlight("<?php   $a = 1; ");
+
+    checkSymbolExistence(1, 7, false);   // (blank)
+    checkSymbolExistence(1, 8, true);    // $
+    checkSymbolExistence(1, 9, true);    // a
+    checkSymbolExistence(1, 10, false);  // (blank)
 
-    assertThat(data).hasSize(1);
-    assertData(data.get(0), 8, 10, Collections.<Integer>emptyList());
```

Comment:
@ivandalbosco we were asserting that this symbol has no references. It's not done now